File: Battleship/Battleship/Models/Requests/UpdateShipPositionRequest.cs


using Battleship.Logic.Enums;
using Battleship.Logic.ViewModels;

namespace Battleship.Models.Requests
{
    public class UpdateShipPositionRequest
    {
        public int GameId { get; set; }
        public int PlayerId { get; set; }
        public ShipType ShipType { get; set; }
        public CoordinatesViewModel StartPosition { get; set; }
        public ShipOrientationType ShipOrientationType { get; set; }
    }
}

File: Battleship/Battleship/ClientApp/src/app/models/requests/update-ship-position-request.ts
import { CoordinatesModel } from '../coordinates-model';
import { ShipOrientationType } from '../enums/ship-orientation-type';
import { ShipType } from '../enums/ship-type';

export class UpdateShipPositionRequest {
  constructor (
    public gameId: number,
    public playerId: number,
    public shipType: ShipType,
    public startPosition: CoordinatesModel,
    public shipOrientationType: ShipOrientationType
  ) {}
}
